lib/utilities/get-package-base-name.js

```diff
@@ -1,12 +1,34 @@
 'use strict';
 
+var npa = require('npm-package-arg');
+
+function fromHostedName(name) {
+  var packageParts, namePart, nameAndVersionParts;
+
+  packageParts = name.split('/');
+  namePart = packageParts[(packageParts.length - 1)];
+  nameAndVersionParts = namePart.split(/[#]/);
+
+  return nameAndVersionParts[0].replace(/.git$/, '');
```

Comment:
The npm methodology for testing this has already given up and not figured out a name. Next steps is that it would try to install it and see if there is a name specified in a package.json inside the .tgz or directory, or bail.

If we care a ton we can implement that. Thoughts?